gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/modes/Modes.java

```diff
@@ -0,0 +1,142 @@
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModifyCheckApiOption;
+import com.palantir.gradle.suppressibleerrorprone.modes.common.ModifyCheckApiOption.CombinedValue;
+import com.palantir.gradle.suppressibleerrorprone.modes.interferences.DisableModeInterference;
+import com.palantir.gradle.suppressibleerrorprone.modes.interferences.RemovingAndSuppressingInterference;
+import com.palantir.gradle.suppressibleerrorprone.modes.interferences.SuppressingAndApplyingInterference;
+import com.palantir.gradle.suppressibleerrorprone.modes.modes.ApplyMode;
+import com.palantir.gradle.suppressibleerrorprone.modes.modes.DisableMode;
+import com.palantir.gradle.suppressibleerrorprone.modes.modes.RemoveRolloutMode;
+import com.palantir.gradle.suppressibleerrorprone.modes.modes.SuppressMode;
+import com.palantir.gradle.suppressibleerrorprone.modes.modes.TimingsMode;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import one.util.streamex.EntryStream;
+import one.util.streamex.StreamEx;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+/**
+ * Use this class to determine the final combination of {@link Mode}s and {@link CommonOptions} for the various
+ * flags given to suppressible-error-prone.
+ */
+public abstract class Modes {
+    @Inject
+    protected abstract ProviderFactory getProviderFactory();
+
+    private final Map<ModeName, Mode> modes = Map.of(
+            ModeName.APPLY, new ApplyMode(),
+            ModeName.DISABLE, new DisableMode(),
+            ModeName.REMOVE_ROLLOUT, new RemoveRolloutMode(),
+            ModeName.SUPPRESS, new SuppressMode(),
+            ModeName.TIMINGS, new TimingsMode());
+
+    private final Set<ModeInterference> interferences = Set.of(
+            new DisableModeInterference(),
+            new RemovingAndSuppressingInterference(),
+            new SuppressingAndApplyingInterference());
+
+    public final CombinedValue modifyCheckApi() {
+        return ModifyCheckApiOption.combine(modesEnabledAndFlagValues().keySet().stream()
+                .map(modes::get)
+                .map(Mode::modifyCheckApi)
+                .collect(Collectors.toSet()));
+    }
+
+    public final CommonOptions commonOptionsFor(JavaCompile javaCompile) {
+        Map<ModeName, Optional<String>> modeNameToFlagValue = modesEnabledAndFlagValues();
+
+        Map<Interference, ModeInterference> interferingModes = StreamEx.of(interferences)
+                .mapToEntry(interference -> interference.interferesWith(modeNameToFlagValue.keySet()))
+                .filterValues(modeInterferenceResult -> modeInterferenceResult instanceof Interference)
+                .mapValues(modeInterferenceResult -> (Interference) modeInterferenceResult)
```

Comment:
you can replace this with:

```suggestion
                .selectValues(Interference.class)
```